internal/kubeclient/indexer.go

```diff
@@ -293,6 +293,9 @@ func FreightByWarehouseIndexer(obj client.Object) []string {
 	if freight.Origin.Kind == kargoapi.FreightOriginKindWarehouse {
 		return []string{freight.Origin.Name}
 	}
+	if freight.Warehouse != "" { // nolint: staticcheck
+		return []string{freight.Warehouse} // nolint: staticcheck
+	}
```

Comment:
This ensures the Warehouse reconciler can find Freight that hasn't been upgraded yet when it goes looking to see if Freight equivalent to one its about to create already exists.